pkg/osbuild/erofs_stage.go

```diff
@@ -0,0 +1,24 @@
+package osbuild
+
+type ErofsCompression struct {
+	Method string `json:"method"`
+	Level  *int   `json:"level,omitempty"`
+}
+
+type ErofsStageOptions struct {
+	Filename string `json:"filename"`
+
+	Compression     *ErofsCompression `json:"compression,omitempty"`
+	ExtendedOptions []string          `json:"options,omitempty"`
+	ClusterSize     *int              `json:"cluster-size,omitempty"`
+}
+
+func (ErofsStageOptions) isStageOptions() {}
+
+func NewErofsStage(options *ErofsStageOptions, inputPipeline string) *Stage {
```

Comment:
(nitpick) I personally like accompanying test for the stages, I put a tiny one in https://github.com/osbuild/images/commit/a368a6dd7c76183b1dad2f8148c863f6f28fa0a7 and I'm happy to do it as a followup. It may seem silly and its definitely busywork but I see (small) benefits because a) it ensures the json tags are right initially and next time something is added at least the omitempty will not forgoten :) and b) that there is a ready-made example how this will look like in json in a real manifest. But totally optional of course, I can propose it via followup or we ignore it :)

Reply:
Ugh, I totally forgot to add that, thanks!